tests/nested/manual/split-refresh/task.yaml

```diff
@@ -0,0 +1,224 @@
+  go build -o muinstaller "$TESTSLIB"/muinstaller/main.go
+
+  # create fake disk for the installer to work on
+  truncate --size=5G fake-disk.img
+  loop_device=$(losetup --show -f ./fake-disk.img)
+  # and "install" the current seed to the fake disk
+  ./muinstaller "$LABEL" "$loop_device" "$TESTSLIB"/muinstaller/mk-classic-rootfs.sh
+  # validate that the fake installer created the expected partitions
+  sfdisk -d "$loop_device" > fdisk_output
+  MATCH "${loop_device}p1 .* name=\"BIOS Boot\""   < fdisk_output
+  # TODO: the real MVP hybrid device will not contain a ubuntu-seed
+  #       partition (needs a different gadget)
+  MATCH "${loop_device}p2 .* name=\"EFI System partition\"" < fdisk_output
+  MATCH "${loop_device}p3 .* name=\"ubuntu-boot\"" < fdisk_output
+  MATCH "${loop_device}p4 .* name=\"ubuntu-save\"" < fdisk_output
+  MATCH "${loop_device}p5 .* name=\"ubuntu-data\"" < fdisk_output
+
+  # image partitions are not mounted anymore
+  for d in ubuntu-seed ubuntu-boot ubuntu-data ubuntu-save; do
+      test -d /run/mnt/"$d"
+      not mountpoint /run/mnt/"$d"
+  done
+
+  # mount image to inspect data
+  mount -o ro "${loop_device}"p3 /run/mnt/ubuntu-boot
+  mount -o ro "${loop_device}"p5 /run/mnt/ubuntu-data
+
+  # seed is populated
+  test -d /run/mnt/ubuntu-data/var/lib/snapd/seed/systems/"$LABEL"
+  # rootfs is there
+  test -x /run/mnt/ubuntu-data/usr/lib/systemd/systemd
+  # ensure not "ubuntu-data/system-data" is generated, this is a dir only
+  # used on core and should not be there on classic
+  not test -d /run/mnt/ubuntu-data/system-data
+  # TODO: ensure we don't have this
+  #not test -d /run/mnt/ubuntu-data/_writable_defaults
+  # and the boot assets are in the right place
+  test -e /run/mnt/ubuntu-boot/EFI/ubuntu/kernel.efi
+  test -e /run/mnt/ubuntu-boot/EFI/ubuntu/grubenv
+  test -e /run/mnt/ubuntu-boot/EFI/boot/grubx64.efi
+  # and we have a modenv in the image
+  MATCH "mode=run" < /run/mnt/ubuntu-data/var/lib/snapd/modeenv
+  MATCH "classic=true" < /run/mnt/ubuntu-data/var/lib/snapd/modeenv
+  umount /run/mnt/ubuntu-{boot,data}
+
+  # HACK: better to change nested_start_vm() to take an image name
+  # Note that use "core" here as the boot is so close to core that classic
+  # does not work
+  IMAGE_NAME="$(nested_get_image_name core)"
+  mv fake-disk.img "$NESTED_IMAGES_DIR/$IMAGE_NAME"
+
+  # boot into the created image
+  # Note that use "core" here as the boot is so close to core that classic
+  # does not work
+  touch "$NESTED_IMAGES_DIR/$IMAGE_NAME.configured"
+  tests.nested create-vm core
+
+  remote.exec "cat /etc/os-release" | MATCH 'NAME="Ubuntu"'
+  remote.exec "snap changes" | MATCH "Done.* Initialize system state"
+
+  echo "Install the required apps and check for expected snaps"
+  remote.exec "sudo snap install test-snapd-tools-core22 test-snapd-sh"
+
+  for snap in snapd core22 pc pc-kernel test-snapd-tools-core22 test-snapd-sh; do
+    remote.exec "snap list" | MATCH "$snap"
+  done
```

Comment:
This could be `remote.exec "snap list snapd core22 pc pc-kernel test-snapd-tools-core22 test-snapd-sh"`

Reply:
Running `snap list snapd non-existent-snap` returns 0 so that won't work as a check that the snaps are installed